### ISSUE TITLE
[ECS/Circuit Breaker]: ability to define for which errors the CB apply

### ISSUE THREAD

**Issue by herbertgoto** [aws/containers-roadmap#1273]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
What do you want us to build?
The ability to define the errors that will trigger a Circuit Breaker rollback. 

**Which service(s) is this request for?**
This could be Fargate, ECS, EKS, ECR
ECS

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
Circuit Breaker rollbacks for any kind of unhealthy deployments and I would like to specify for which ones to do it. In the case of lack of resources available in the cluster, I want my deployment to be retried until my cluster scales accordingly instead of rolling back. 

**Are you currently working around this issue?**
How are you currently solving this problem?
I have to act upon a lack of resources event and retry the deployment after a while. 

**Additional context**
Anything else we should know?

**Attachments**
If you think you might have additional information that you'd like to include via an attachment, please do - we'll take a look. (Remember to remove any personally-identifiable information.)

**Comment by dezren39** [comment]
I am also interested in a similar concept, even if exit code = 0, I'd like to be able to set a minimum duration for the container to be up. Like healthCheckGrace, except reverse. If N was 30 and the container was up for 25 seconds before exit 0, then it would increment fail count even as the task is marked 'exited gracefully..' or however.

**Comment by eliot1019** [comment]
We are using ECS Circuit Breaker on most of our ECS services.

For services on Fargate, we've seen Circuit Breaker kick in and fail deployments when AWS has no Fargate capacity available (our services are configured for multi az).

The ECS deployment events will show:
`service xxx was unable to place a task. Capacity is unavailable at this time. Please try again later or in a different availability zone...`

This requires us to have to manually redeploy the service even if nothing about the new deployment configuration and code is wrong.

We'd love to be able to exclude these placement/capacity issues from the Circuit Breaker threshold and for ECS to automatically retry these deployments like it already does when CB is off.

**Comment by erwang01** [comment]
Yes particularly helpful would be to exclude spot instances not being available since its understandable that not all spot instances can be allocated at once and will be filled in as they become available.
